python_modules/dagster-webserver/dagster_webserver/webserver.py

```diff
@@ -231,15 +230,31 @@ def index_html_endpoint(self, request: Request):
                 """)
 
     def build_static_routes(self):
-        def next_file_response(file_path):
-            with open(file_path, encoding="utf8") as f:
-                content = f.read().replace(
-                    "BUILDTIME_ASSETPREFIX_REPLACE_ME", f"{self._app_path_prefix}"
-                )
-                return Response(content=content, media_type=guess_type(file_path)[0])
+        temp_dir = tempfile.TemporaryDirectory() if self._uses_app_path_prefix else None
+
+        def next_file_response(path, file_path):
+            # Create the full temporary directory path, including subdirectories
+            temp_dir_path = os.path.join(temp_dir.name, *os.path.dirname(path).split(os.path.sep))
```

Comment:
extremely subtle that we use `temp_dir.name` even though `temp_dir` can be `None` based on disconnected `self._uses_app_path_prefix` gating

Reply:
yeah the path that uses it `next_file_response` is also gated by the same variable so I'll remove the gating here.